includes/class-wc-payments.php

```diff
@@ -283,8 +283,10 @@ public static function init() {
 			include_once __DIR__ . '/multi-currency/wc-payments-multi-currency.php';
 		}
 
-		// // Load platform checkout save user section if feature is enabled.
-		add_action( 'woocommerce_cart_loaded_from_session', [ __CLASS__, 'init_platform_checkout' ] );
+		// Load platform checkout save user section if feature is enabled.
+		// It needs to be after calculate totals due to subscriptions with trial
+		// that can change the WC()->cart->needs_payment() value.
+		add_action( 'woocommerce_after_calculate_totals', [ __CLASS__, 'init_platform_checkout' ] );
```

Comment:
I'm a little uncertain about potential consequences of this approach for a couple of reasons. While `woocommerce_cart_loaded_from_session` happens exactly once as a hook on `wp_loaded`, `woocommerce_after_calculate_totals` happens every time the cart's `calculate_totals` is called. As a result:
- Might it be possible that `calculate_totals` gets called multiple times in a single request, and as a result we end up taking actions like enqueuing scripts / setting up tracking multiple times?
- Since `calculate_totals` happens later in a request than `wp_loaded`, can we think of anything that might require `init_platform_checkout` to have been called earlier that would break now that it's later?

I think this will require some additional investigation/testing to verify.